spec/models/mail_server_log_spec.rb

```diff
@@ -331,21 +331,56 @@
 
   describe '#delivery_status' do
 
-    context 'using the :exim MTA' do
+    context 'if there is a stored value' do
+      let(:log) do
+        MailServerLog.new(:line => "log text",
+                          :delivery_status => 'bounce_arrival')
+      end
+
+      it 'returns the stored value' do
+        status = MailServerLog::EximDeliveryStatus.new(:bounce_arrival)
+        expect(log.delivery_status).to eq(status)
+      end
+
+      it 'does not look at the line text' do
+        expect(log).to_not receive(:line)
+        log.delivery_status
+      end
+
+    end
+
+    context 'there is not a stored value' do
+
+      it 'attempt to parse the line text' do
+        log = MailServerLog.new(:line => "hi")
+        expect(log).to receive(:line)
+        log.delivery_status
```

Comment:
This is testing the internals a bit much IMO.

I'd give it real data to actually do the work:

```ruby
log = MailServerLog.new(:line => "…<=…")
expect(log.delivery_status).
  to eq(MailServerLog::EximDeliveryStatus.new(:message_arrival))
```